### PR TITLE
Issue #174: Migration of HouseController to OpenApi yaml schema

### LINKED ISSUE #174: Migration of HouseController to OpenApi yaml schema
This task is a subtask for #169.

The steps that need to be done:
- convert all endpoints from controller to yaml file
- check that endpoints are working, in the same way, like before the change
- check the tests

An example can be found in PR #171.

### PR DESCRIPTION
**Pull request by edwin9870**
## 🚀 Description

Added OpenAPI spec for HouseController

## 📄 Motivation and Context

Closes #174 
## 🧪 How Has This Been Tested?

- Test cases are passing
- Verified functionalities via running locally Home-Web UI

## 📦 Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [X] My code follows the code style of this project(Do your best to follow code styles. If none apply just skip this).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java [hunk 1]
```diff
@@ -147,12 +147,13 @@ void getHouseDetailsNotExists() {
   void listAllMembersOfHouse() {
     // given
     Set<HouseMember> testHouseMembers = TestUtils.HouseMemberHelpers.getTestHouseMembers(TEST_HOUSE_MEMBERS_COUNT);
-    Set<ListHouseMembersResponse.HouseMember> testHouseMemberDetails = testHouseMembers.stream()
-        .map(member -> new ListHouseMembersResponse.HouseMember(member.getMemberId(),
-            member.getName()))
+    Set<com.myhome.model.HouseMember> testHouseMemberDetails = testHouseMembers.stream()
+        .map(member -> new com.myhome.model.HouseMember().memberId(member.getMemberId()).name(member.getName())
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: Can we reformat it in a more readable way?

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java [hunk 2]
```diff
@@ -194,17 +195,15 @@ void addHouseMembers() {
...
         .collect(Collectors.toSet());
 
-    AddHouseMemberRequest request = new AddHouseMemberRequest(testMembersDto);
+    AddHouseMemberRequest request = new AddHouseMemberRequest().members(testMembersDto);
 
-    Set<AddHouseMemberResponse.HouseMember> addedMembers = testMembers.stream()
-        .map(member -> new AddHouseMemberResponse.HouseMember(member.getMemberId(),
-            member.getName()))
+    Set<com.myhome.model.HouseMember> addedMembers = testMembers.stream()
+        .map(member -> new com.myhome.model.HouseMember().memberId(member.getMemberId()).name(member.getName()))
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: Can we reformat it in a more readable way?

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java [hunk 3]
```diff
@@ -234,17 +233,18 @@ void addHouseMembersNoMembersAdded() {
...
-    AddHouseMemberRequest request = new AddHouseMemberRequest(testMembersDto);
+    AddHouseMemberRequest request = new AddHouseMemberRequest().members(testMembersDto);
 
-    Set<AddHouseMemberResponse.HouseMember> addedMembers = testMembers.stream()
-        .map(member -> new AddHouseMemberResponse.HouseMember(member.getMemberId(),
-            member.getName()))
+    Set<com.myhome.model.HouseMember> addedMembers = testMembers.stream()
+        .map(member -> new com.myhome.model.HouseMember()
+                .memberId(member.getMemberId())
+              .name(member.getName()))
```
Comment (mslowiak) [inline service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: I guess something is incorrect with padding

File: api/src/main/resources/api.yaml [hunk 4]
```diff
@@ -435,6 +436,141 @@ paths:
...
+        '200':
+          description: if community exists
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/GetHouseDetailsResponse'
+            application/xml:
+              schema:
+                $ref: '#/components/schemas/GetHouseDetailsResponse'
+        '404':
```
Comment (mslowiak) [inline api/src/main/resources/api.yaml]: `/houses` will never return 404, please delete that case from yaml

**Review by mslowiak — CHANGES_REQUESTED** [review]
Some changes needed but looks good

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java (hunk 1 above)
Comment (edwin9870) [reply service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: Done!

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java (hunk 2 above)
Comment (edwin9870) [reply service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: Done!

File: service/src/test/java/com/myhome/controllers/HouseControllerTest.java (hunk 3 above)
Comment (edwin9870) [reply service/src/test/java/com/myhome/controllers/HouseControllerTest.java]: Fixed

File: api/src/main/resources/api.yaml (hunk 4 above)
Comment (edwin9870) [reply api/src/main/resources/api.yaml]: Fixed!

**Pull request merged**